management-cluster-access/README.md

```diff
@@ -0,0 +1,119 @@
+- Since it is already integrated, it makes sense to keep it for emergencies while there is no alternative.
+- If we want to support this for a longer time we need to improve security.
+- We could support this as fallback method in `opsctl login` by managing access to lastpass.com or calling the `lastpass` CLI to ease operations.
+
+### Introduce a second identity provider for SSO
+
+Dex supports a wide array of identity providers. Customers are already using many of them. At the present moment, giant swarm staff can only login using github.
+However, we could add more providers.
+
+- We favor SSO when it comes to identity management for human engineers.
+- Could be introduced for workload cluster access as well.
+- Using e.g. azure AD would make sense since customers are using it.
+- We might be able to simplify account automation when using another identity provider than github.
+- We would still have a dependency on dex and this method will not completely remove the single point of failure.
+- Adding another provider might mean individual setup for each existing configuration. (e.g. in github individual apps are needed) We should avoid this if possible since it already does not scale.
+- While dex allows an array of connectors, we currently use a fixed naming for our single `giantswarm` connector. We need to revisit what relies on this convention to ensure the second connector is equal.
+- Regardless of other means auf authentication being implemented, adding another dex idp would be beneficial.
+
+### Service account tokens
+
+Other than client certificates, we could also look at service accounts as a means of managing access.
+
+- Service accounts can be rotated so they carry less risk in that regard than client certificates.
+- This should not be a standard access method for human engineers.
+- We would need to discover how to implement this. A dedicated controller for managing service account access is thinkable.
+
+### Other access methods
+
+- To be discussed
+
+### Related stories
+
+This discussion is focused on management cluster access by GS staff and removing the single point of failure here should remain the goal.
+However, our chosen solution for other access related problems has impact on the issue of management cluster access by giant swarm staff. We should aim to resuse the same mechanisms for workload clusters and management clusters as much as possible. Likewise we should strive to dogfood mechanisms we offer to our customers.
+On the other hand, we should be conscious of differences between access for automation and controllers versus access for humans. 
+Related problems include:
+
+- Access to management clusters for controllers
+- Access to management clusters for customers
+- Access to workload clusters for controllers
+- Access to workload clusters by gs staff
+- Access to workload clusters by customers
+
+## Next steps
+
+### Adding azure AD as a second identity provider for SSO
```

Comment:
out of curiosity: why not using `google` as second OIDC provider? Everyone at GS has a google-account per default. Azure AD is something additional?

Reply:
The Google connector in Dex is marked as "alpha", while the Azure one is mature, and we don't have any experience with it.

We already maintain user accounts in Azure AD.